adam/learner/learner_utils.py

```diff
@@ -231,3 +231,14 @@ def covers_entire_utterance(
     # This assumes the slots and the non-slot elements are non-overlapping,
     # which is true for how we construct them.
     return num_covered_tokens == sized_tokens
+
+
+def no_object_alignment_in_fixed_strings(
+    bound_surface_template: SurfaceTemplateBoundToSemanticNodes,
+    language_concept_alignment: LanguageConceptAlignment,
+) -> bool:
+    for (node, _) in language_concept_alignment.node_to_language_span.items():
+        if node.concept.debug_string in bound_surface_template.surface_template.elements:
```

Comment:
Will this approach fail for Chinese? Would it be better to examine the aligned token positions directly?

Reply:
In our current unit tests, this doesn't fail for Chinese. I don't believe it should fail because the `debug_string` of a concept is derived from the language which was originally aligned with the concept. -- However, I should be able to run a more thorough test by running the experiment pipeline over the learner in Chinese. I'll do so and report back.

Reply:
Got it.  In any case, though, is the information available here to do the task by comparing token indices?  That is less brittle, since we really want to minimize the amount we rely on `debug_string` (since it is, after all, for debugging and not for logic)

Reply:
Hm, I think I may be able to do so given we assume that the entire utterance is covered before checking this condition. Without that assumption, there could be potential token offsets to account for.